### PR TITLE
[now deploy] Determine platform version through `Dockerfile` and `package.json`

### PR DESCRIPTION
**Pull request by AndyBitz**
When the users platform version is set to `1`, but the project is missing a `Dockerfile` and / or a `package.json` with a `start` script, or if it was not set otherwise, then we'll default to `2` instead.

### CODE REVIEW

**Review by leo — CHANGES_REQUESTED** [review]
Does this consider:

1. CLI option for setting platform version to 1
2. static 1.0 deployments with just now.json
3. `version` set to `1`

**Comment by AndyBitz** [comment]
@leo 

> CLI option for setting platform version to 1
> version set to 1 (I guess in `now.json`)

When either of those is true then we don't change the version

> static 1.0 deployments with just now.json

Right, the `type` is not yet considered, I'll add that

File: src/commands/deploy/index.js [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
+}
+
+async function hasPkgStartScript(cwd) {
+  const pkg = await readPackage(join(cwd, 'package.json')).catch(() => null);
+
+  if (!pkg || pkg instanceof Error) {
+    return false;
+  }
+
+  const { scripts } = pkg;
```
Comment (TooTallNate) [inline src/commands/deploy/index.js]:
```suggestion
  const { scripts = {} } = pkg;
```

**Review by TooTallNate — CHANGES_REQUESTED** [review]
Needs tests

**Comment by AndyBitz** [comment]
@TooTallNate added unit tests, integration tests would require the user to set their platform version to 1 through the API, but this is not allowed for new users (or test users that are just created for the tests).

File: src/util/prefer-v2-deployment.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  hasDockerfile,
+  pkg,
+  localConfig
+}: {
+  hasDockerfile: boolean,
+  pkg: Package | CantParseJSONFile | null,
+  localConfig: Config | undefined
+}): Promise<null | string> {
+  if (localConfig && localConfig.version) {
+    // We will prefere anything that is set here
```
Comment (TooTallNate) [inline src/util/prefer-v2-deployment.ts]:
```suggestion
    // We will prefer anything that is set here
```

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**